pyqtgraph/colormap.py

```diff
@@ -1,3 +1,7 @@
+__all__ = [
+    # 'listMaps', 'get',
+    'getFromMatplotlib', 'getFromColorcet', 'makeMonochrome', 'modulatedBarData', 'ColorMap']
```

Comment:
I think 
`__all__ = ['ColorMap']`
should be sufficient here. The function names were chosen to live in a separate colormap namespace and don't make much sense as e.g. `pg.getFromColorcet`  (or even `pg.get`)

To me it seems cleaner to have
`pg.colormap.get('CET1')`
rather than 
`pg.getColormap('CET1')`
and it doesn't require any more typing either.

Reply:
For the stuff in functions.py we should maybe start by changing the documentation, and conceptionally move the more abstract functions to live in `pg.functions`.

My personal preference for deciding what should live in the top-level `pg.` namespace would be to ask "Would I need this when doing manual data visualization in an interactive notebook?" That is where short, easily memorized calls are most relevant. From that point of view, the pg. namespace should hold things like `mkColor()`, but not `arrayToQPath()`.

As far as I can tell, the use of the more abstract functions throughout the library is typically something like
```python
import pg.functions as fn
fn.eq(a,b)
```
and maybe the documentation should encourage that use, too.

Right now, we also have functionality like `affineSliceCoords` in the pg. namespace, even though this is not even in the documentation.

This is obviously a policy discussion that requires more input from more people, but from the technical side of things:
Would there be a convenient way to add deprecation messages for the use of selected abstract functions directly from the `pg.` namespace  (while keeping the access through `pg.functions` indisturbed)?

Reply:
I have implemented the deprecation following the pointer to https://www.python.org/dev/peps/pep-0562/ from @ntjess